### PR TITLE
Test Coverage improved for data products

### PR DESCRIPTION
**Pull request by utk14**
The PR consists of test coverage for the EdDataProducts repo. PFA the updated test coverage:
Screenshot 2021-04-01 at 5 35 19 PM

### CODE REVIEW

File: data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala [hunk 1]
```diff
@@ -221,20 +220,44 @@ class TestUserInfoExhaustJob extends BaseReportSpec with MockFactory with BaseRe
...
+    val strConfig = """{"search":{"type":"none"},"model":"org.sunbird.analytics.exhaust.collection.UserInfoExhaustJob","modelParams":{"store":"local","mode":"standalone", "batchId": "batch-001","batchFilter":["TPD"],"sparkElasticsearchConnectionHost":"localhost","sparkRedisConnectionHost":"localhost","sparkUserDbRedisIndex":"0","sparkCassandraConnectionHost":"localhost","sparkUserDbRedisPort":6381,"fromDate":"","toDate":"","storageContainer":""},"parallelization":8,"appName":"UserInfo Exhaust"}"""
     val jobConfig = JSONUtils.deserialize[JobConfig](strConfig)
     implicit val config = jobConfig
 
     UserInfoExhaustJob.execute()
+
+    val outputDir = "response-exhaust"
+    val batch1 = "batch-001"
+    val filePath = UserInfoExhaustJob.getFilePath(batch1)
```
Comment (manjudr) [inline data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala:233]: I think you should pass the `requestId` as well here. Please verify once. 
Take the latest code once.

File: data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala [hunk 2]
```diff
@@ -249,14 +272,25 @@ class TestUserInfoExhaustJob extends BaseReportSpec with MockFactory with BaseRe
...
     val jobRequestArr = Array(jobRequest)
-    val outputLocation = AppConf.getConfig("collection.exhaust.store.prefix")
     val storageConfig = StorageConfig("local", "", outputLocation)
     implicit val conf = spark.sparkContext.hadoopConfiguration
 
     UserInfoExhaustJob.saveRequests(storageConfig, jobRequestArr)
 
   }
 
+  it should "generate the report without modelParams present" in {
```
Comment (manjudr) [inline data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala:282]: Please add the assertions.

File: data-products/src/test/scala/org/sunbird/analytics/util/TestUnirestUtil.scala [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+    val url = "https:/httpbin.org/post?type=test";
     val request = Map("popularity" -> 1);
-    val response  = UnirestUtil.post(url, JSONUtils.serialize(request))
-    val resError = JSONUtils.deserialize[PostError](response)
-    resError.url shouldBe("https://httpbin.org/post?type=test")
+
+    try {
+      UnirestUtil.post(url, JSONUtils.serialize(request))
+    } catch {
+      case ex: UnirestException => println("exmess: " + ex.getMessage)
```
Comment (manjudr) [inline data-products/src/test/scala/org/sunbird/analytics/util/TestUnirestUtil.scala:16]: Please add proper error logs. `exmess` invalid string in the log

**Review by manjudr — CHANGES_REQUESTED** [review]

File: data-products/src/test/scala/org/sunbird/analytics/util/TestUnirestUtil.scala (hunk 3 above)
Comment (utk14) [reply data-products/src/test/scala/org/sunbird/analytics/util/TestUnirestUtil.scala:16]: Done

File: data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala (hunk 1 above)
Comment (utk14) [reply data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala:234]: Done

File: data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala (hunk 2 above)
Comment (utk14) [reply data-products/src/test/scala/org/sunbird/analytics/exhaust/TestUserInfoExhaustJob.scala:284]: Done

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Pull request merged**